src/common/src/array/serial_array.rs

```diff
@@ -84,3 +92,68 @@ impl<'a> ScalarRef<'a> for Serial {
         self.0.hash(state)
     }
 }
+
+#[derive(Debug)]
+pub struct SerialArrayBuilder(PrimitiveArrayBuilder<i64>);
+
+#[derive(Debug, Clone)]
+pub struct SerialArray(PrimitiveArray<i64>);
```

Comment:
What about `impl PrimitiveArrayItemType for Serial` and then `pub type SerialArray = PrimitiveArray<Serial>;` (see `NaiveDateArray`)?

Reply:
Cool, i'll try it.